Pcap++/src/PcapFilter.cpp

```diff
@@ -170,17 +169,15 @@ void IPFilter::convertToIPAddressWithLen(std::string& ipAddrmodified, int& len)
 	}
 	else if (ipAddr.isIPv4())
 	{
-		const IPv4Address& ip4Addr = ipAddr.getIPv4();
-		uint32_t addrAsInt = ip4Addr.toUInt();
-		uint32_t mask = ((uint32_t)-1) >> ((sizeof(uint32_t)*8)-m_Len);
+		uint32_t addrAsInt = ipAddr.getIPv4().toUInt();
+		uint32_t mask = (uint32_t)(-1) >> ((sizeof(uint32_t) * 8) - m_Len);
 		addrAsInt &= mask;
 		ipAddrmodified = IPv4Address(addrAsInt).toString();
 	}
 	else // IPv6
 	{
-		const IPv6Address& ip6Addr = ipAddr.getIPv6();
-		uint8_t* addrAsArr; size_t addrLen;
-		ip6Addr.copyTo(&addrAsArr, addrLen);
+		uint8_t addrAsArr[16];
+		ipAddr.getIPv6().copyTo(addrAsArr);
 		uint64_t addrLowerBytes = (long)addrAsArr;

```

Comment:
Is this code right? A pointer is being converted to uint64_t...

Reply:
I think, this method has an error for IPv6 addresses because `addrLowerBytes` and `addrHigherBytes` variables are not copied back to `addrAsArr`

Reply:
yes I think you're right, this is probably not covered in the unit-test. Would you mind fixing it?

Reply:
I guess `IPFilter` does not support `IPv6`. I have found the following `@todo Add IPv6 filtering support` in header.

Reply:
I think you're right 😃 I wrote this code a long time ago, I can't remember why I added this piece of code. Could you please remove it?

Reply:
Code has been removed